dist/s_clang-scan.diff

```diff
@@ -14,14 +14,20 @@ src/conn/conn_capacity.c:291:5: warning: Value stored to 'capacity' is never rea
     capacity = steal_capacity = 0;
     ^          ~~~~~~~~~~~~~~~~~~
 1 warning generated.
-src/reconcile/rec_col.c:1050:23: warning: Null pointer passed as an argument to a 'nonnull' parameter
-                      memcmp(last.value->data, data, size) == 0)) {
-                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-1 warnings generated.
+src/reconcile/rec_col.c:1063:25: warning: Null pointer passed as an argument to a 'nonnull' parameter
+                        memcmp(last.value->data, data, size) == 0))) {
+                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+1 warning generated.
 In file included from src/reconcile/rec_write.c:9:
 In file included from ./src/include/wt_internal.h:421:
 ./src/include/mutex.i:184:16: warning: Null pointer passed as an argument to a 'nonnull' parameter
     if ((ret = pthread_mutex_unlock(&t->lock)) != 0)
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1 warning generated.
+In file included from src/txn/txn_rollback_to_stable.c:9:
+In file included from ./src/include/wt_internal.h:421:
+./src/include/mutex.i:158:13: warning: Null pointer passed as an argument to a 'nonnull' parameter
+    return (pthread_mutex_trylock(&t->lock));
+            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+1 warning generated.
```

Comment:
Again, you didn't change `wt_internal.h` nor did you change any locking near line 9 in `txn_rollback_to_stable.c` so I'm not sure why this changed. Admittedly you did change `txn_rollback_to_stable.c` and these errors are mysterious, but I'd still like to understand this.

Reply:
Yes, I didn't change anything in `wt_internal.h` but a similar error like above is already masked, so I added this also.